### ISSUE TITLE
Add example in for looping through parameters and passing them as environment variables

### ISSUE THREAD

**Issue by GenesisCoast** [MicrosoftDocs/azure-devops-docs#7851]
After some extensive testing we've found some additional functionaility that should be documented.

For example you can use the for loop to iterate through parameters:

```
parameters:
  test_parameter: test_value

steps: 
- ${{ each parameter in parameters }}:
  - script: echo ${{ parameter.Key }} # Outputs 'test_parameter'
  - script: echo ${{ parameter.Value }} # Outputs 'test_value'
```

It should also be documented that parameters can be passed to tasks such as `PowerShell` and `AzurePowerShell` using the environment variables:

```
parameters:
  test_parameter: test_value

steps: 
- task: PowerShell@2
  env:
    ${{ each parameter in parameters }}:
      ${{ parameter.Key }}: ${{ parameter.Value }}
  inputs:
    filePath: test_script.ps1
    pwsh: true
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 790318bb-8220-3241-4ca7-73351074492f
* Version Independent ID: db1da9db-3694-779b-17aa-1ed67fcecf86
* Content: [Use runtime and type-safe parameters - Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/runtime-parameters?view=azure-devops)
* Content Source: [docs/pipelines/process/runtime-parameters.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/pipelines/process/runtime-parameters.md)
* Product: **devops**
* Technology: **devops-cicd-process**
* GitHub Login: @juliakm
* Microsoft Alias: **jukullam**

**Comment by juliakm** [comment]
@GenesisCoast Thanks for the suggestion! You can do loop through variables for string, number, and boolean types. Looks like it doesn’t work for object, step, or stepList. I'm add this as a tip on the page.

**Comment by MrOwen** [comment]
For anyone else coming across this issue, I had troubles accessing `parameter.Value`. Instead, `parameter.Value` seemed to be an object with a few other properties and the key that holds the actual value is `lit`. So to access the value, I used `${{ parameter.Value.lit) }}`. It seems like this is not intended but I've tested several scenarios and this fix seems to be the key to resolving my issues.

@juliakm Are you able to reach out to engineering to verify this?
